File: code/523/solution.go
package main /*{{{*/

import (
	"fmt"
)

func main() {
	ans := float64(0)
	base := float64(2)
	for i := 2; i <= 30; i++ {
		ans += (base - 1) / float64(i)
		base *= 2
	}
	fmt.Printf("%.2f\n", ans)
}

File: code/523/solution.cpp
#include <iostream>  /*{{{*/
using namespace std; /*}}}*/
int main() {
    double ans  = 0;
    double base = 2;
    for (int i = 2; i <= 30; i++) {
        ans += (base - 1) / i;
        base *= 2;
    }
    printf("%.2f\n", ans);
    return 0;
}
